src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -1828,4 +1828,58 @@ private void doTestIssue777_2(HttpClient client, String uri, String expectation,
 		            .expectErrorMessage(expectation)
 		            .verify(Duration.ofSeconds(30));
 	}
+
+	@Test
+	public void testConnectionIdleTimeFixedPool() {
+		ConnectionProvider provider = ConnectionProvider.fixed("test", 1, 100, Duration.ofMillis(10));
+		ChannelId[] ids = doTestConnectionIdleTime(provider);
+		assertThat(ids[0]).isNotEqualTo(ids[1]);
+	}
+
+	@Test
+	public void testConnectionIdleTimeElasticPool() {
+		ConnectionProvider provider = ConnectionProvider.elastic("test", Duration.ofMillis(10));
+		ChannelId[] ids = doTestConnectionIdleTime(provider);
+		assertThat(ids[0]).isNotEqualTo(ids[1]);
+	}
+
+	@Test
+	public void testConnectionNoIdleTimeFixedPool() {
+		ConnectionProvider provider = ConnectionProvider.fixed("test", 1, 100);
+		ChannelId[] ids = doTestConnectionIdleTime(provider);
+		assertThat(ids[0]).isEqualTo(ids[1]);
+	}
+
+	@Test
+	public void testConnectionNoIdleTimeElasticPool() {
+		ConnectionProvider provider = ConnectionProvider.elastic("test");
+		ChannelId[] ids = doTestConnectionIdleTime(provider);
+		assertThat(ids[0]).isEqualTo(ids[1]);
+	}
+
+	private ChannelId[] doTestConnectionIdleTime(ConnectionProvider provider) {
+		DisposableServer server =
+				HttpServer.create()
+				          .port(0)
+				          .wiretap(true)
+				          .handle((req, res) -> res.sendString(Mono.just("hello")))
+				          .bindNow();
+
+		Flux<ChannelId> id = createHttpClientForContextWithAddress(server, provider)
+		                       .get()
+		                       .uri("/")
+		                       .responseConnection((res, conn) -> Mono.just(conn.channel().id())
+		                                                              .delayUntil(ch -> conn.inbound().receive()));
+
+		ChannelId id1 = id.blockLast(Duration.ofSeconds(30));
+		Mono.delay(Duration.ofMillis(30)).block();
```

Comment:
is that on purpose? that feels weirder than using `Thread.sleep(30)` or `Awaitility` 😉

Reply:
;)
changed to `Thread.sleep(30)`